src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -560,6 +579,12 @@ private void NoteCaptured(Symbol variable)
                 _usedVariables.Add(local);
             }
 
+            if (variable is ParameterSymbol parameter && CurrentSymbol is SynthesizedRecordConstructor)
```

Comment:
>CurrentSymbol is SynthesizedRecordConstructor [](start = 57, length = 45)

It looks like a parameter can be used in a lambda. Is this condition going to work for this scenario?

Reply:
Added tests for lambdas. Thanks for the scenario!

---
In reply to: [512814777](https://github.com/dotnet/roslyn/pull/48895#discussion_r512814777) [](ancestors = 512814777)